Source/Wikitivity.Agent/WikitivityAgentHelper.cs

```diff
@@ -101,28 +214,41 @@ public List<WikitivityUploadsAgent.DataObtainedSingleRequestObject> PopulateArti
 
 		public static string GetInstanceUrl(IAgentHelper helper, AgentBase aB)

```

Comment:
Don't pass in an AgentBase.  Rather you could pass in a IProgress<String> implementation that can message for you.
Then if this code was ever called outside of an agent, they could write their own implementation of the progress and not need to update the class